src/lib/storage/dictionary_segment/dictionary_encoder.hpp

```diff
@@ -58,19 +58,48 @@ class DictionaryEncoder : public SegmentEncoder<DictionaryEncoder<Encoding>> {
       }
     });
 
-    auto dictionary = std::make_shared<pmr_vector<T>>(values.cbegin(), values.cend(), allocator);
+    auto dictionary = std::make_shared<pmr_vector<T>>(dense_values.cbegin(), dense_values.cend(), allocator);
     std::sort(dictionary->begin(), dictionary->end());
     dictionary->erase(std::unique(dictionary->begin(), dictionary->end()), dictionary->cend());
     dictionary->shrink_to_fit();
 
+    const auto null_value_id = static_cast<uint32_t>(dictionary->size());
+
+    auto create_compressed_attribute_vector = [&](const auto& local_dictionary) {
```

Comment:
Rename the dictionary from line 61 to `temp_dictionary` and use `dictionary` here?